ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -50,192 +52,144 @@ TestListBuckets(const Aws::S3Crt::ClientConfiguration &config)
 int
 TestListObjects(const Aws::S3Crt::ClientConfiguration &config)
 {
-    int ret = 0;
     S3Connection conn(config);
 
     /* Temporary workaround to get a bucket to use. */
-    std::string firstBucket;
     std::vector<std::string> buckets;
-
+    int ret;
     if (ret = conn.ListBuckets(buckets) != 0)
-        goto err;
-    if (!buckets.empty()) {
-        firstBucket = buckets.at(0);
+        return (0);
+    if (buckets.empty()) {
+        std::cout << "No buckets found in AWS account." << std::endl;
+        return (1);
+    }
+
+    std::string firstBucket = buckets.at(0);
 
-        std::vector<std::string> objects;
-        uint32_t countp;
+    std::vector<std::string> objects;
+    uint32_t countp;
 
-        /* No matching objects. */
-        if (ret = conn.ListObjects(firstBucket, "test_list_objects_", objects, countp) != 0)
-            goto err;
-        if (objects.size() != countp || countp != 0) {
-            ret = 1;
-            goto err;
-        }
+    /* No matching objects. */
+    if (ret = conn.ListObjects(firstBucket, "test_list_objects_", objects, countp) != 0)
+        return (ret);
+    if (objects.size() != countp || countp != 0) {
+        return (1);
+    }
 
-        /* Create file to prepare for test. */
-        if (!static_cast<bool>(std::ofstream("test_list_objects.txt").put('.'))) {
-            std::cerr << "Error creating file." << std::endl;
-            ret = 1;
-            goto err;
-        }
+    /* Create file to prepare for test. */
+    if (!static_cast<bool>(std::ofstream("test_list_objects.txt").put('.'))) {
+        std::cerr << "Error creating file." << std::endl;
+        return (1);
+    }
 
-        /* Put objects to prepare for test. */
-        for (int i = 0; i < 20; i++) {
-            if (ret = conn.PutObject(firstBucket, "test_list_objects_" + std::to_string(i) + ".txt",
-                        "test_list_objects.txt") != 0)
-                goto err;
-        }
+    /*
+     * totalObjects: Number of test objects added to bucket. expectedResult: Expected number of
```

Comment:
np: You don't need to specify `totalObjects` and `expectedResult`  in the comment. Separate the comment in two and add them above the variable they refer to.